js/app/application.js

```diff
@@ -141,6 +146,9 @@ const Application = new Knowledge.Class({
     vfunc_startup: function () {
         this.parent();
         let engine = Engine.get_default();
+        if (engine instanceof MoltresEngine.MoltresEngine)
```

Comment:
I'm not sure I totally understood what you were requesting, but have refactored the code that was in `vfunc_startup()` into a separate `_check_for_content()` function.